vision_model/plugin_base.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, Dict, Tuple
+import typing
+import torch
+from .trainer_callback_interface import TrainerCallbackInterface
+
+
+if typing.TYPE_CHECKING:
+    from .model_base import ModelBase
+
+
+class PluginBase:
+    """Callbacks to modify the training behavior."""
```

Comment:
Are we changing the "callback" to "plugin"?

Reply:
yes, I thought the name "callback" might limit the future extensions. will fix the comment at line 12.